contracts/Core/RewardManager.sol

```diff
@@ -282,75 +158,50 @@ contract RewardManager is Initializable, ACL, RewardStorage {
+        uint256 previousAge = thisStaker.age;
         uint256 epochLastRevealed = thisStaker.epochLastRevealed;
 
         Structs.Block memory _block = blockManager.getBlock(epochLastRevealed);
 
-        uint256[] memory lowerCutoffsLastEpoch = _block.lowerCutoffs;
-        uint256[] memory higherCutoffsLastEpoch = _block.higherCutoffs;
+        uint256[] memory mediansLastEpoch = _block.medians;
 
-        if (lowerCutoffsLastEpoch.length > 0) {
+        if (mediansLastEpoch.length > 0) {
             uint256 penalty = 0;
-            for (uint256 i = 0; i < lowerCutoffsLastEpoch.length; i++) {
+            for (uint256 i = 0; i < mediansLastEpoch.length; i++) {
                 uint256 voteLastEpoch =
                     voteManager
                         .getVote(epochLastRevealed, thisStaker.id, i)
                         .value;
-                uint256 lowerCutoffLastEpoch = lowerCutoffsLastEpoch[i];
-                uint256 higherCutoffLastEpoch = higherCutoffsLastEpoch[i];
+                uint256 medianLastEpoch = mediansLastEpoch[i];
 
-                if (
-                    (voteLastEpoch < lowerCutoffLastEpoch) ||
-                    (voteLastEpoch > higherCutoffLastEpoch)
-                ) {
-                    penalty =
-                        penalty +
-                        (previousStake / parameters.exposureDenominator());
+                if (voteLastEpoch > medianLastEpoch) {
+                    penalty = penalty +
+                    (previousAge * (voteLastEpoch - medianLastEpoch)**2)
+                    /medianLastEpoch**2;
+                } else {
+                    penalty = penalty +
+                    (previousAge*(medianLastEpoch - voteLastEpoch)**2)
+                    /medianLastEpoch**2;
                 }
             }
 
-            if (penalty > 0) {
-                penalty = (penalty > previousStake) ? previousStake : penalty;
-                stakeManager.setStakerStake(
-                    thisStaker.id,
-                    (previousStake - (penalty)),
-                    "Voting Penalty",
-                    epoch
-                );
-                uint256 prevRewardPool = rewardPool;
-                rewardPool = rewardPool + (penalty);
-                emit RewardPoolChange(
-                    epoch,
-                    prevRewardPool,
-                    rewardPool,
-                    block.timestamp
-                );
-            } else {
-                //no penalty. only reward
-                uint256 prevStakeGettingReward = stakeGettingReward;
-                stakeGettingReward = stakeGettingReward + (previousStake); //*(1 - y);
-                emit StakeGettingRewardChange(
-                    epoch,
-                    prevStakeGettingReward,
-                    stakeGettingReward,
-                    block.timestamp
-                );
-            }
+            uint256 newAge = (previousAge + 10000 - (penalty));
```

Comment:
What is 10000 here?

Reply:
age is increased by 10,000 every epoch. This is to have enough precision since we don't have decimals in solidity.